app/src/features/Columns/ColumnSelect.tsx

```diff
@@ -62,13 +71,25 @@ const ColumnSelects = ({
     }
   );
 
-  const { table, column, owner: ownerId } = pendingColumn;
-  const selectedOwner = credentialOwners?.find(({ id }) => id === ownerId);
+  const {
+    data: mapping,
+    isLoading: mappingLoading,
+  } = useApiResourcesRetrieveQuery(
+    { id: mappingId ?? "" },
+    { skip: !mappingId }
+  );
+
+  const { table: pendingColumnTable, column, owner: ownerId } = pendingColumn;
+  const table = (pendingColumnTable || mapping?.primary_key_table) ?? undefined;
+  const selectedOwner = ownerId
+    ? credentialOwners?.find(({ id }) => id === ownerId)
+    : credentialOwners?.find(({ id }) => id === mapping?.primary_key_owner);
```

Comment:
It shouldn't be the `ColumnSelect` job to set the mapping table.

The props `pendingColumn` itself should be initialized with the mapping table instead, because "tricking" with the input value won't set the pendingColumn `table` value.

What happens here is that the input has the right value, but not `pendingColumn`.

Reply:
You can check the problem when setting a column.

If you don't change the table and set a column value, there won't be any request made to create a `Column` django object (because `pendingColumn` won't have any `owner` / `table` value set).

But if you change the `table` value first, and then set a `column` value, you'll see that a request is made to create a `Column` django object